src/Compilers/CSharp/Portable/Emitter/Model/SymbolAdapter.cs

```diff
@@ -126,5 +126,23 @@ internal IEnumerable<CSharpAttributeData> GetCustomAttributesToEmit(ModuleCompil
                 }
             }
         }
+
+        internal Cci.TypeReferenceWithAttributes GetTypeRefWithAttributes(TypeSymbol type, Cci.ITypeReference typeRef)
+        {
+            ImmutableArray<Cci.ICustomAttribute> attrs;
+            if (type.ContainsTuple())
+            {
+                var attr = DeclaringCompilation.SynthesizeTupleNamesAttributeOpt(type);
```

Comment:
Perhaps pass `Compilation` explicitly as a parameter - currently it's less obvious which `Symbol` receiver to call this method on. Then, the method could be defined in `TypeSymbolExtensions`.